router/batchrouter/asyncdestinationmanager/sftp/utils.go

```diff
@@ -0,0 +1,315 @@
+package sftp
+
+import (
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	jsoniter "github.com/json-iterator/go"
+
+	sftp "github.com/rudderlabs/rudder-go-kit/sftp"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
+
+// createSSHConfig creates SSH configuration based on destination
+func createSSHConfig(destination *backendconfig.DestinationT) (*sftp.SSHConfig, error) {
+	destinationConfigJson, err := json.Marshal(destination.Config)
+	if err != nil {
+		return nil, err
+	}
+
+	var config destConfig
+	if err := json.Unmarshal(destinationConfigJson, &config); err != nil {
+		return nil, err
```

Comment:
I guess typecasting is better for performance than doing a marshal and unmarshal